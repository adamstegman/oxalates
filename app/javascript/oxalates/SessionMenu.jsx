import React from 'react';
import PropTypes from 'prop-types';

import { SessionForm } from './SessionForm';
import { SessionLogin } from './SessionLogin';
import { SessionLogout } from './SessionLogout';

export class SessionMenu extends React.Component {
  render() {
    const {
      authenticated,
      authenticating,
      error,
      password,
      onChangePassword,
      onStartAuthentication,
      onStopAuthentication,
    } = this.props;

    if (authenticating) {
      return <SessionForm error={error}
                          password={password}
                          onChangePassword={onChangePassword}
                          onStopAuthentication={onStopAuthentication} />;
    }

    if (authenticated) {
      return <SessionLogout onChangePassword={onChangePassword} />;
    }

    return <SessionLogin onStartAuthentication={onStartAuthentication} />;
  }
};

SessionMenu.propTypes = {
  authenticated: PropTypes.bool,
  authenticating: PropTypes.bool,
  error: PropTypes.string,
  password: PropTypes.string,
  onChangePassword: PropTypes.func,
  onStartAuthentication: PropTypes.func,
  onStopAuthentication: PropTypes.func,
};
