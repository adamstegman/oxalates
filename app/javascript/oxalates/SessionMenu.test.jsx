import React from 'react';
import renderer from 'react-test-renderer';

import { SessionMenu } from './SessionMenu';

test('SessionMenu renders the unauthenticated session menu', () => {
  const component = renderer.create(
    <SessionMenu authenticated={false} authenticating={false} password="" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SessionMenu renders the authentication form', () => {
  const component = renderer.create(
    <SessionMenu authenticated={false} authenticating={true} password="pass" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SessionMenu renders the failed authentication form', () => {
  const error = 'some error';
  const component = renderer.create(
    <SessionMenu authenticated={false} authenticating={true} password="" error={error} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SessionMenu renders the authenticated session menu', () => {
  const component = renderer.create(
    <SessionMenu authenticated={true} authenticating={false} password="password" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
