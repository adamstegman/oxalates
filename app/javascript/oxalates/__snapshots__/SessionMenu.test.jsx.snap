// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionMenu renders the authenticated session menu 1`] = `
<button
  className="action logout"
  onClick={[Function]}
  type="button"
>
  Log out
</button>
`;

exports[`SessionMenu renders the authentication form 1`] = `
<form
  className="session"
  onSubmit={[Function]}
>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    autoFocus={true}
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value="pass"
  />
  <input
    type="submit"
    value="Log in"
  />
</form>
`;

exports[`SessionMenu renders the failed authentication form 1`] = `
<form
  className="session"
  onSubmit={[Function]}
>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    autoFocus={true}
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <input
    type="submit"
    value="Log in"
  />
  <p
    className="item"
  >
    some error
  </p>
</form>
`;

exports[`SessionMenu renders the unauthenticated session menu 1`] = `
<button
  className="action login"
  type="button"
>
  Log in
</button>
`;
